File: ProjetoProgSofApp/php/src/model/Reserva.php
<?php

namespace model;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'tb_reserva')]
class Reserva extends GenericModel
{
    #[ORM\Column(type: 'date', name: 'data_reserva', nullable: true)]
    private $dataReserva;

    #[ORM\ManyToOne(targetEntity: Cliente::class, fetch: 'EAGER')]
    #[ORM\JoinColumn(name: 'cliente_id', nullable: true)]
    private $cliente;

    #[ORM\ManyToOne(targetEntity: PacoteDeTurismo::class, fetch: 'EAGER')]
    #[ORM\JoinColumn(name: 'pacote_id', nullable: true)]
    private $pacote;

    public function getDataReserva()
    {
        return $this->dataReserva;
    }

    public function setDataReserva($dataReserva): void
    {
        $this->dataReserva = $dataReserva;
    }

    public function getCliente()
    {
        return $this->cliente;
    }

    public function setCliente($cliente): void
    {
        $this->cliente = $cliente;
    }

    public function getPacote()
    {
        return $this->pacote;
    }

    public function setPacote($pacote): void
    {
        $this->pacote = $pacote;
    }
}

File: ProjetoProgSofApp/src/main/java/br/edu/ifpr/bsi/ProjetoProgSofApp/models/Reserva.java
package br.edu.ifpr.bsi.ProjetoProgSofApp.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "tb_reserva")
public class Reserva extends GenericModel {

    @Column(name = "data_reserva")
    private String dataReserva;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "pacote_id")
    private PacoteDeTurismo pacote;
}